### DISCUSSION TITLE
Codespaces does not respect workspaceFolder option in devcontainer.json

### DISCUSSION THREAD

**Discussion by Adjective-Object** [community/community#11923] [Codespaces] [answered]
My org's repo is hosted in Azure Devops, and our github repo is a condespaces config plus some scripts to allow us to use codespaces against that repo.

The way this works right now is that we create a destination directory for the repo in our Dockerfile, then clone the Azure Devops repo in a postCreate script.

We would like the "start codespace" button to launch the codespace in that cloned repo's directory.

At present, the codespace always launches against the shim repo's workspace (`/workspace/shim-repo`) rather than the path in `workspaceFolder`.

We have [workspaceMount and workspaceFolder](https://code.visualstudio.com/docs/remote/devcontainerjson-reference) set in our devcontainer.json

Devcontainer.json
```json
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vscode/master/extensions/configuration-editing/schemas/devContainer.schema.json",
  "name": "ADO in in github codespaces",
  "build": {
    "dockerfile": "./Dockerfile",
    "args": {
      "USERNAME": "vscode"
    }
  },
  "remoteUser": "vscode",
  "settings": {
    "terminal.integrated.shell.linux": "/bin/bash"
  },
  "workspaceMount": "/",
  "workspaceFolder": "/workspaces/ado-repo",
  "extensions": [],
  "remoteEnv": {
    "ADO_REPO_URL": "(omitted)",
    "VM_REPO_PATH": "/workspaces/ado-repo",
    "USER_NPMRC_PATH": "(omitted)",
    "ADO_NPM_FEED": "(omitted)"
  },
  "postCreateCommand": "/bin/bash .codespaces/post-create.sh",
  "postStartCommand": "/bin/bash .codespaces/post-start.sh",
  "forwardPorts": [3000, 30001]
}

```

**Comment by joshspicer** [comment]
Internally, we use a repo forked from https://github.com/vsls-contrib/ado-in-codespaces to contribute to certain pieces of codespaces that live in ADO.  I think taking a look at the devcontainer.json in this repo may be helpful!

**Reply by Adjective-Object** [reply]
We're using the same template as a base repo internally at Outlook. as far as I can tell from that repo definition, the vscode workspace will still boot in the shim repo, and the user has to manually execute the `cddefault` alias to bring them to the checked-out ado repo.

(That is, unless codespaces has some logic around reading `$CODESPACE_DEFAULT_PATH` from the env or something similar?)

I talked to clantz internally as well and from our discussion it seemed like support for changing the workspace default 
codespace path was not yet shipped, but I may be mistaken. Would love to be wrong about this!

**Comment by Denympc** [comment]
Pada Sabtu, 05 Maret 2022, Max ***@***.***> menulis:

> We're using the same template as a base repo internally at Outlook. as far
> as I can tell from that repo definition, the vscode workspace will still
> boot in the shim repo, and the user has to manually execute the cddefault
> alias to bring them to the checked-out ado repo.
>
> (unless codespaces has some logic around reading $CODESPACE_DEFAULT_PATH
> from the env that I'm missing)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/github/feedback/discussions/11923#discussioncomment-2297733>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AX2RH347CS77R3ZJ576HIUTU6J2N5ANCNFSM5PCDLZMQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
null

**Comment by davetapley** [comment]
Confirmed still happening, [minimal repro here](https://github.com/JEFuller/vscode-dev-containers/commit/7782e6121d7100b1c585e53cdfc7ede0b4bca3f0) forked from https://github.com/microsoft/vscode-dev-containers/.

Works fine in local devcontainer ✅ 
```
node ➜ /foo (main) $ 
```

Is ignored in Codespace ❌ 
```
@davetapley ➜ /workspaces/vscode-dev-containers (main) $ 
```

---

Apparently was to be fixed in late 2020, so not sure if broken again or never working 🤔 
https://github.com/orgs/community/discussions/25449#discussioncomment-3247942


---

Ref for myself as part of my quest to free up space on the prompt: https://github.com/microsoft/vscode-dev-containers/issues/1196#issuecomment-1343553053

**Comment by OneCyrus** [comment, marked as answer]
just hit the same issue. would be great to have a consistent workflow in codespaces and local

**Reply by OneCyrus** [reply]
@jkeech
any insights into the progress here? is this still on the radar? would be great to close this feature gap.

looks like it was on the table in 2020.
https://github.com/orgs/community/discussions/25449
![image](https://user-images.githubusercontent.com/7747800/218685298-815274ac-6645-40ca-a04b-77be752bcef8.png)

**Reply by jkeech** [reply]
`workspaceFolder` is fully supported now! Let us know if you are hitting any issues.

**Reply by OneCyrus** [reply]
awesome! just tried one and it looks to work as expected.

**Reply by solesensei** [reply]
Still have the issue. Now working with multiple `devcontainers.json`. In logs `devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/my_repo` even after recreation with `"workspaceFolder": "/workspace"`

**Comment by JamesHutchison** [comment]
I encountered this issue with the following flow:
- Open an existing codespace
- Create a new nested configuration
- Open the nested configuration from within VS Code
- Breaks
- Fix config and rebuild
- Stuck on workspace folder of parent / root of project. workspaceFolder is not respected.
- Stopping and starting the codespace does NOT resolve the issue

Committing the changes and creating a fresh codespace gave the correct folder.

**Reply by weihongliang233** [reply]
I spent a night on this problem ……

Committing the changes and recreating a codespace instance works.

![image](https://github.com/community/community/assets/52309635/b9e3b97e-be17-4cf8-8fde-791b437f7aec)

Modifying `devcontainer.json: workspaceFolder` from inside an existing codespace does not work. We have to push the modification and choose the correct branch when creating the codespace instance.
